comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala

```diff
@@ -105,28 +116,25 @@ class TcpTransportLayer[
                     case p if p.isInternalServerError =>
                       Left(internalCommunicationError("crap"))
                 })
-                .pure[F]
+                .pure[Task]
     } yield pmErr
 
-  val local: F[PeerNode] = src.pure[F]
+  // TODO: rename to sendAndForget
+  def send(peer: PeerNode, msg: ProtocolMessage): Task[CommErr[Unit]] =
+    Task
+      .racePair(innerSend(peer, TLRequest(msg.proto.some)), Task.unit)
```

Comment:
As per my understanding of `Task.racePair` it doesn't make sens to race `Task.unit` . Why did you choose to do that?

Reply:
Please distinguish between `Task.race` and `Task.racePair`. `Task.race` cancells the losing `Task`.

Reply:
Won't `innerSend(…).attempt` achieve the same result? and `Task.timeout` is [not implemented with racing a `Task.unit`](https://github.com/monix/monix/blob/master/monix-eval/shared/src/main/scala/monix/eval/Task.scala#L1803).

Reply:
No, `Task.attempt` turns a `Task[A]` into `Task[Either[Throwable, A]]`. Such a `Task` can't result in an `Error`.
**Send and forget** means that we are scheduling a message for sending and don't wait for the execution nor the response. We don't care if the message actually gets sent, delivered and handled by the remote peer. However, there are some edge cases, f.e. we need to provide a grace period (not included in this PR) on shutdown to deliver the `Disconnect` message to our peers.